llpc/util/llpcElfWriter.cpp

```diff
@@ -823,6 +834,144 @@ Result ElfWriter<Elf>::getSectionTextShader(const char *name, const void **data,
   return Result::Success;
 }
 
+// =====================================================================================================================
+// Get the start pos of relocs used by Ps in the reloc section
+//
+// @param relocSection : Reloc section
+// @param psIsaOffset : Ps ISA offset in elf binary
+template <class Elf> size_t ElfWriter<Elf>::getRelocPsOffset(const SectionBuffer *relocSection, size_t psIsaOffset) {
+  size_t psStartPos = 0;
+
+  const size_t relocCount = static_cast<size_t>(relocSection->secHead.sh_size / relocSection->secHead.sh_entsize);
+
+  auto relocs = reinterpret_cast<typename Elf::Reloc *>(const_cast<uint8_t *>(relocSection->data));
```

Comment:
Similar code appears elsewhere. Can you make a helper function that given a `relocSection` returns the pointer to `Elf::Reloc *`?